spark/src/main/scala/geotrellis/spark/join/VectorJoin.scala

```diff
@@ -36,31 +36,29 @@ object VectorJoin {
 
       Iterator(rtree)
     }, preservesPartitioning = true)
-      .zipWithIndex
-      .map({ case (v, k) => (k, v) })
       .cache
```

Comment:
This shouldn't cache, no longer necessary (and also we as library developers should avoid caching - that's a client concern that could have negative performance impact based on your setup)